src/_posts/languages/php/2000-01-01-extensions.md

```diff
@@ -0,0 +1,148 @@
+---
+title: Managing PHP Extensions
+nav: Managing PHP Extensions
+modified_at: 2023-06-02 16:00:00
+tags: php
+index: 3
+---
+
+Applications might require native PHP extensions which are usually written in
+C. Consequently, they need to be compiled as shared libraries (`.so` files) to
+be used by PHP.
+
+Some of them are pre-installed and shipped by default with the deployed version
+of PHP. Others are dynamically installed if specified in the project's
+`composer.json` file.
+
+## Pre-Installed Extensions
+
+The following extensions are **available and enabled** by default with the
+installed version of PHP. You don't need to add them in your `composer.json`
+file (but you can).
+
+* Databases: [`mysql`](https://www.php.net/manual/en/book.mysql.php),
+[`mysqli`](https://www.php.net/manual/en/book.mysqli.php),
+[`pgsql`](https://www.php.net/manual/en/book.pgsql.php),
+[`pdo-mysql`](https://www.php.net/manual/en/ref.pdo-mysql.php),
+[`pdo-pgsql`](https://www.php.net/manual/en/ref.pdo-pgsql.php),
+* Compression: [`bz2`](https://www.php.net/manual/en/book.bzip2.php),
+[`zlib`](https://www.php.net/manual/en/book.zlib.php),
+[`zip`](https://www.php.net/manual/en/book.zip.php)
+* Web services: [`soap`](https://www.php.net/manual/en/book.soap.php)
+* XML manipulation: [`xmlreader`](https://www.php.net/manual/en/mbstring.installation.php)
+* String encoding: [`mbstring`](https://www.php.net/manual/en/mbstring.installation.php)
+* Process control: [`pcntl`](https://www.php.net/manual/en/book.pcntl.php)
+* Socket management: [`sockets`](https://www.php.net/manual/en/book.sockets.php)
+* Math functions: [`bcmath`](https://www.php.net/manual/book.bc.php)
+* Images manipulation: [`gd`](https://www.php.net/manual/book.image.php)
+* Intl (Internationalization): [`intl`](https://www.php.net/manual/book.intl.php)
+
+## Working With PECL Extensions
+
+All [PECL extensions](https://pecl.php.net/) are available. If the extension
+you need is not in the above list of pre-installed extensions, it will be
+compiled during the *build* phase of your deployment.
+
+To enable a native PHP extension, add it in the `require` block of your
+`composer.json`, like so (notice the `ext-` prefix):
+
+```json
+{
+  "require": {
+    "ext-mongodb": "*",
+    "ext-imagick": "*",
+    ...
+  }
+}
+```
+
+Once done, don't forget to generate the `composer.lock` file and to commit it
+to your codebase.
```

Comment:
question: I don't think adding an extension changes the `composer.lock`. I think it's only an instruction for the platform.